src/indicator/nmofono/vpn/vpn-manager.cpp

```diff
@@ -278,6 +280,63 @@ QString VpnManager::addConnection(VpnConnection::Type type)
     return uuid;
 }
 
+QString VpnManager::importConnection(QString filepath, VpnConnection::Type type)
```

Comment:
Please adjust the added code below to match the coding style of the rest of the indicator. I think the `filepath` argument should also be taken as `const QString& filepath` instead.  It would probably also be good for the argument order to match that of the function which is calling into this one, for consistency.

As another point, all of the `g_print()` statements in this method should be removed. Many also don't make so much sense any more, as we're not looping through all plugins, and are mostly extraneous things only useful for the example code. It looks like there might still be a couple memory leaks, and it might be a good idea to wrap some of the libnm types as a `std::shared_ptr` to improve the memory management and make it easier to just `throw runtime_error();` in the cases where errors occur. That would simplify the code, and make it more suitable for future modification if needed, as well.

Reply:
Alright, I'm moving the filepath type, though I didn't find the order match you are looking for, the function calling is sending filepath first, and the vpn connection type afterwards, see below:
```
QString uuid = p.d->m_vpnManager->importConnection(
                    filepath,
                    static_cast<VpnConnection::Type>(type)
            );
```

Reply:
aw, I see it now, it is the function parameters of the calling function:
```
QDBusObjectPath PrivateService::ImportVpnConnection(int type, const QString &filepath) {
[...]
```

Reply:
The new version addresses these comments, except for the std::shared_ptr/throw runtime_error() comment for now. I'll take a look what I can do in that direction.

Reply:
Hi, sorry, it seems that we can't do the std::shared_ptr since the NMConnection and NMVpnEditorPlugin are incomplete types enforcing the necessity of having a destructor, see the output below:
```
In file included from /usr/include/c++/5/bits/shared_ptr.h:52:0,
                 from /usr/include/c++/5/memory:82,
                 from /usr/include/unity/util/DefinesPtrs.h:22,
                 from /home/phablet/indicator-network/src/indicator/nmofono/connection/active-connection.h:25,
                 from /home/phablet/indicator-network/src/indicator/nmofono/connection/active-connection-manager.h:24,
                 from /home/phablet/indicator-network/src/indicator/nmofono/vpn/vpn-manager.h:22,
                 from /home/phablet/indicator-network/src/indicator/nmofono/vpn/vpn-manager.cpp:21:
/usr/include/c++/5/bits/shared_ptr_base.h: In instantiation of ‘std::__shared_ptr<_Tp, _Lp>::__shared_ptr(_Tp1*) [with _Tp1 = _NMVpnEditorPlugin; _Tp = _NMVpnEditorPlugin; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]’:
/usr/include/c++/5/bits/shared_ptr.h:117:32:   required from ‘std::shared_ptr<_Tp>::shared_ptr(_Tp1*) [with _Tp1 = _NMVpnEditorPlugin; _Tp = _NMVpnEditorPlugin]’
/home/phablet/indicator-network/src/indicator/nmofono/vpn/vpn-manager.cpp:305:5:   required from here
/usr/include/c++/5/bits/shared_ptr_base.h:886:25: error: invalid application of ‘sizeof’ to incomplete type ‘_NMVpnEditorPlugin’
    static_assert( sizeof(_Tp1) > 0, "incomplete type" );
                         ^
/usr/include/c++/5/bits/shared_ptr_base.h: In instantiation of ‘std::__shared_ptr<_Tp, _Lp>::__shared_ptr(_Tp1*) [with _Tp1 = _NMConnection; _Tp = _NMConnection; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]’:
/usr/include/c++/5/bits/shared_ptr.h:117:32:   required from ‘std::shared_ptr<_Tp>::shared_ptr(_Tp1*) [with _Tp1 = _NMConnection; _Tp = _NMConnection]’
/home/phablet/indicator-network/src/indicator/nmofono/vpn/vpn-manager.cpp:315:5:   required from here
/usr/include/c++/5/bits/shared_ptr_base.h:886:25: error: invalid application of ‘sizeof’ to incomplete type ‘_NMConnection’
src/indicator/CMakeFiles/indicator-network-service-static.dir/build.make:598: recipe for target 'src/indicator/CMakeFiles/indicator-network-service-static.dir/nmofono/vpn/vpn-manager.cpp.o' failed
```

Reply:
I pushed now a last update to my patch which at least wraps the NMVpnPluginInfo type, which seems
to be the only one that is not affected by an incomplete type. For now, it seems the best we can do. Let
me know if you find it otherwise.

Reply:
Hi @dobey, can I assume this is fine and we are good to go with the patch? Is there any more comments/
improvements you'd like me to pursue?

Thanks for the comments!

José.